lib/sqlalchemy/dialects/postgresql/base.py

```diff
@@ -3643,20 +3643,152 @@ def get_multi_columns(
 
         return columns.items()
 
-    def _get_columns_info(self, rows, domains, enums, schema):
-        array_type_pattern = re.compile(r"\[\]$")
-        attype_pattern = re.compile(r"\(.*\)")
-        charlen_pattern = re.compile(r"\(([\d,]+)\)")
-        args_pattern = re.compile(r"\((.*)\)")
-        args_split_pattern = re.compile(r"\s*,\s*")
-
-        def _handle_array_type(attype):
-            return (
-                # strip '[]' from integer[], etc.
-                array_type_pattern.sub("", attype),
-                attype.endswith("[]"),
+    _format_type_args_pattern = re.compile(r"\((.*)\)")
+    _format_array_spec_pattern = re.compile(r"((?:\[\])*)$")
+
+    def _reflect_type(
+        self,
+        format_type: Optional[str],
+        domains: dict[str, ReflectedDomain],
+        enums: dict[str, ReflectedEnum],
+        type_description: Optional[str] = None,
```

Comment:
**Federico Caselli** (CaselIT) wrote:

this is always provided so I don't think there is any reason to make it an optional str

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5030

Reply:
It's always provided, but is unused except to add additional information to errors. The function could technically be called without it. It could arguably default to `"<unknown source>"` or something.

Reply:
it's an internal function likely ever only called by a single place, so it's probably fine to make it required